common/MessageQueue.hpp

```diff
@@ -82,7 +82,7 @@ class MessageQueue
     /// Thread safe remove_if.
     void remove_if(const std::function<bool(const Payload&)>& pred)
     {
-        std::remove_if(_queue.begin(), _queue.end(), pred);
+        (void)std::remove_if(_queue.begin(), _queue.end(), pred);
```

Comment:
I think this use of std::remove_if is really dubious without a paired std::erase to actually remove the shuffled unwanted element. Luckily it seems that this MessageQueue::remove_if isn't actually used and this is dead code. I think we should remove the dead code in a follow up